src/index.js

```diff
@@ -41,6 +41,39 @@ client.on('threadCreate', async (thread, newlyCreated) => {
   )
 })
 
+client.on('threadUpdate', async (oldThread, newThread) => {
+  const isForumChannel = constants.forumChannels.some(
+    it => it.id === newThread.parentId
+  )
+
+  if (!isForumChannel) return
+  if (!oldThread.archived) return
+  if (newThread.archived) return
+
+  /** @type {import('discord.js').ThreadChannel<true>} */
+  const thread = newThread
+  const logger = eventLogger.createChild('threadUpdate')
+
+  logger.info(`"${thread.name}" (${thread.id}) has been reopened.`)
+
+  const guild = thread.guild
+  const entry = await guild
+    .fetchAuditLogs({
+      type: AuditLogEvent.ThreadUpdate,
+      limit: 1,
+    })
+    .then(it => it.entries.first())
+  const unarchived = entry.changes.some(
+    it => it.key === 'archived' && it.old && !it.new
+  )
+
+  if (!entry) return
+  if (entry.target.id !== thread.id) return
+  if (!unarchived) return
+
+  await thread.send(`${entry.executor}がスレッドを再開しました。`)
```

Comment:
該当の監査ログが見つからなかった場合でも、スレッドが再開されたことだけでもわかった方が良いと思います。
```suggestion
  if (entry && entry.target.id === thread.id && unarchived) {
    await thread.send(`${entry.executor}がスレッドを再開しました。`)
  } else {
    await thread.send(`スレッドが再開されました。`)
  }
```